modules/core/src/main/java/org/apache/ignite/internal/IgniteFeatures.java

```diff
@@ -65,7 +65,10 @@
     TX_TRACKING_UPDATE_COUNTER(12),
 
     /** Support new security processor */
-    IGNITE_SECURITY_PROCESSOR(13);
+    IGNITE_SECURITY_PROCESSOR(13),
+
+    /** Support of dereferencing TcpDiscoveryNode field in discovery messages. */
+    TCP_DISCOVERY_MESSAGE_NODE_SERIALIZATION_OPTIMIZATION(14);
```

Comment:
To me this name isn't descriptive enough. What do you think about NODE_COMPACT_REPRESENTATION_OPTIMIZATION?